webapp/src/main/java/life/qbic/security/UserDetailsServiceImpl.java

```diff
@@ -24,7 +25,15 @@ public class UserDetailsServiceImpl implements UserDetailsService {
 
   @Override
   public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
-    var user = userRepository.findByEmail(EmailAddress.from(username));
+    EmailAddress email;
+    // Check if the email address is valid
+    try {
+      email = EmailAddress.from(username);
+    } catch (EmailValidationException e) {
+      throw new UsernameNotFoundException("Cannot find user");
+    }
+    // Then search for a user with the provided email address
+    var user = userRepository.findByEmail(email);
```

Comment:
why is this part of the PR?

Reply:
I think in general it would be nice if an error is thrown that also catching the error is covered